wallet-new/integration/Functions.hs

```diff
@@ -73,10 +73,10 @@ runActionCheck
     -> ActionProbabilities
     -> m WalletState
 runActionCheck walletClient walletState actionProb = do
-    actions <- chooseActions 100 actionProb
+    actions <- chooseActions 10 actionProb
     log $ "Test will run these actions: " <> show (toList actions)
     let client' = hoistClient lift walletClient
-    ws <- execRefT (tryAll (map (runAction client') actions)) walletState
+    ws <- execRefT (tryAll (map (runAction client') actions) <|> pure ()) walletState
```

Comment:
From https://iohk.myjetbrains.com/youtrack/issue/CSL-2480#comment=93-20502 : 

My sentiment is that this doesn't really fix the problem. For example running `[DeleteWallet]` list of actions will now succeed where actually `DeleteWallet` action has been skipped. If someone just made a bug in delete wallet method in API it would make a regression and integration test would be unable to catch it.

With this modification I would at least add a condition for test to succeed is that all actions from list of actions should be successfully run at least once. Do you think this makes sense? Otherwise this test will sometimes succeed with no good reason to succeed.

I can do something like https://github.com/input-output-hk/cardano-sl/blob/267537f31ef597564f1b95e72d52f7e8a40a1dca/wallet-new/integration/Functions.hs#L109 : 
```haskell
   succs <- use successActions
   when (not . null $ ([minBound..maxBound] \\ nub succs) $
      throwM "Test skipped some actions and we don't have a ground to say everything is ok (when not everything has been tested)."
```

Other then that, `tryAll` would (unexpected given its name) unwrup to:
```haskell
((((((action1 *> action2) <|> action2) *> action3) <|> action3) *> action4) <|> action4) ....
```
which shows that only action1 will be tried exactly one time, where all other actions (if certan conditions are met) can be tried two times. This didn't bite us only because:
 * we are running actions in sequence
 * if action is failing, it will be tried again as a next action immediatelly after and most probably it will also fail

That is also the reason why `WalletState.actionsNum` will be greater then list of actions in a list of actions - as some actions are tried and failed multiple times - with no real harm for now (but it might bit us)

So my question is: what should be expected success condition of these tests? Is it:
 * all actions should be run at least once successfully (doesn't have failing invariant)
     - solid condition to me
 * at least one action should be run successfully (doesn't have failing invariant)
     - I find this a weak condition
 * none of actions should have failing invariant
     - this is current condition
     - I find this a weak condition as skipped actions also have no failing conditions (as we didn't test anything)

Summary:
 * what should be successful test condition?
 * do we want to fix `tryAll` which doesn't really work as expected and tries some actions multiple times or we don't care?